diesel_cli/src/infer_schema_internals/pg.rs

```diff
@@ -90,25 +90,46 @@ pub fn get_table_data(
     conn: &mut PgConnection,
     table: &TableName,
     column_sorting: &ColumnSorting,
+    domain_types_enabled: bool,
 ) -> QueryResult<Vec<ColumnInformation>> {
     use self::information_schema::columns::dsl::*;
+    use diesel::sql_types::{Nullable, SingleValue, Text};
 
     let schema_name = match table.schema {
         Some(ref name) => Cow::Borrowed(name),
         None => Cow::Owned(Pg::default_schema(conn)?),
     };
 
+    let (select_name, select_schema) = if domain_types_enabled {
+        define_sql_function!(#[sql_name = "coalesce"] fn coalesce_maybe<T: SingleValue>(x: Nullable<T>, y: Nullable<T>) -> Nullable<T>);
+        define_sql_function!(#[sql_name = "coalesce"] fn coalesce_strict<T: SingleValue>(x: Nullable<T>, y: T) -> T);
+
+        type BoxedTextExpr = Box<dyn BoxableExpression<columns, Pg, SqlType = Text>>;
+        type BoxedMaybeTextExpr = Box<dyn BoxableExpression<columns, Pg, SqlType = Text>>;
```

Comment:
Haha I was like, how does this work without the `Nullable<Text>` here?
The answer is it doesn't 😅
```
 error[E0271]: type mismatch resolving `<coalesce_maybe<Text, domain_schema, Nullable<udt_schema>> as Expression>::SqlType == Text`
   --> diesel_cli/src/infer_schema_internals/pg.rs:112:13
    |
112 |             Box::new(coalesce_maybe(domain_schema, udt_schema.nullable())) as BoxedMaybeTextExpr,
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Text`, found `Nullable<Text>`
    |
    = note: expected struct `diesel::sql_types::Text`
               found struct `diesel::sql_types::Nullable<diesel::sql_types::Text>`
```

Reply:
Yes, I'm sorry! I was in a hurry so I missed it, check this out 8f7a58282e1bdc82b9bec5c1fa45587fd1d4c27c